ddtrace/tracer.py

```diff
@@ -73,42 +78,35 @@ def configure(self, enabled=None, hostname=None, port=None, sampler=None):
         if sampler is not None:
             self.sampler = sampler
 
-    def trace(self, name, service=None, resource=None, span_type=None):
-        """Return a span that will trace an operation called `name`.
+        if context_provider is not None:
+            self._context_provider = context_provider
 
-        :param str name: the name of the operation being traced
-        :param str service: the name of the service being traced. If not set,
-                            it will inherit the service from its parent.
+    def start_span(self, name, child_of=None, service=None, resource=None, span_type=None):
+        """
+        Starts and returns a new ``Span`` representing a unit of work.
+
+        :param str name: the name of the operation being traced.
+        :param object child_of: a Span or a Context instance representing the parent
+                                for this span.
+        :param str service: the name of the service being traced.
         :param str resource: an optional name of the resource being tracked.
         :param str span_type: an optional operation type.
 
-        You must call `finish` on all spans, either directly or with a context
-        manager.
-
-        >>> span = tracer.trace("web.request")
-            try:
-                # do something
-            finally:
-                span.finish()
-        >>> with tracer.trace("web.request") as span:
-            # do something
-
-        Trace will store the created span and subsequent child traces will
-        become it's children.
+        To start a new root span::
 
-        >>> tracer = Tracer()
-        >>> parent = tracer.trace("parent")     # has no parent span
-        >>> child  = tracer.trace("child")      # is a child of a parent
-        >>> child.finish()
-        >>> parent.finish()
-        >>> parent2 = tracer.trace("parent2")   # has no parent span
-        >>> parent2.finish()
+            >>> span = tracer.start_span("web.request")
```

Comment:
Add two examples about:
* how to create a child from a ``Span``
* how to create a child from a ``Context``